### ISSUE TITLE
May I ask why the code that can be called normally in pytorch, but needs to be the same size to be calculated after encryption is used?

### ISSUE THREAD

**Issue by dikjstra** [facebookresearch/CrypTen#406]
When I call torch.nn.CrossEntrypyLoss() to calculate loss in pytorch, parameters of different sizes can be calculated, but when I use crypto to call crypton.nn.CrossEntrypyLoss() to calculate loss, there is an error that different size parameters cannot be calculated , I am very confused, thank you!
![1](https://user-images.githubusercontent.com/90977053/190134934-3ab94bf9-6553-4ada-bd45-dcce3b1b4802.png)
![2](https://user-images.githubusercontent.com/90977053/190134941-637fbcb1-9567-42e3-94ab-390a2bc74bd7.png)
![3](https://user-images.githubusercontent.com/90977053/190134949-2f50dffd-ef70-4c03-bd59-9a1aa05b3947.png)
![4](https://user-images.githubusercontent.com/90977053/190134973-3d937987-f3c3-470d-a18d-39ea378721a6.png)
![错误截图](https://user-images.githubusercontent.com/90977053/190134985-c8a1f612-e2eb-4ad4-a77d-1527f8761801.png)

**Comment by mohammad-alrubaie** [comment]
Can you please provide a minimal working example so we can reproduce the error?

**Comment by knottb** [comment]
This is because the `target` for the `CrossEntropyLoss` is expected to be `one-hot`. The reason for this is that the operation for computing the one-hot vector from an index in the encrypted space is very expensive. Instead, it is better to compute this one-hot vector in plaintext, before it is input into the `CrossEntropyLoss` function.

For reference, given an output with 3 classes, this might look like the following:

Output of softmax:
```
[[0.1, 0.1, 0.8],
 [0.3, 0.5, 0.2],
 ...
 [0.8, 0.1, 0.1]]
```

PyTorch expected target:
```
[2, 1, ..., 0]
```

CrypTen expected target:
```
[[0, 0, 1],
 [0, 1, 0],
 ...
 [1, 0, 0]]
```

**Comment by dikjstra** [comment]
> This is because the `target` for the `CrossEntropyLoss` is expected to be `one-hot`. The reason for this is that the operation for computing the one-hot vector from an index in the encrypted space is very expensive. Instead, it is better to compute this one-hot vector in plaintext, before it is input into the `CrossEntropyLoss` function.
> 
> For reference, given an output with 3 classes, this might look like the following:
> 
> Output of softmax:
> 
> ```
> [[0.1, 0.1, 0.8],
>  [0.3, 0.5, 0.2],
>  ...
>  [0.8, 0.1, 0.1]]
> ```
> 
> PyTorch expected target:
> 
> ```
> [2, 1, ..., 0]
> ```
> 
> CrypTen expected target:
> 
> ```
> [[0, 0, 1],
>  [0, 1, 0],
>  ...
>  [1, 0, 0]]
> ```

Thanks for your answer, I tried it myself and it works!
